### PR TITLE
Added more concrete handling of unexpected events

### PR DESCRIPTION
**Pull request by SEQUOIIA**

### CODE REVIEW

File: src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public Type GetInstanceTypeFor(string eventName)
 
             if (registration == null)
             {
-                throw new MessagingException($"Error! Could not determine \"event instance type\" due to no registration was found for type {eventName}!");
+                throw new MessagingHandlerNotAvailable($"Error! Could not determine \"event instance type\" due to no registration was found for type {eventName}!", eventName);
```
Comment (JacobHeidelbachDFDS) [inline src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs:68]: No tests to prove that we cast MessagingHandlerNotAvailable?

Comment (JacobHeidelbachDFDS) [reply src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs:68]: And should this implementation be ported to other services using "DomainEventRegistry"?

**Comment by janwiberg-dfds** [comment]
Which leads to wondering if it should just be that single change and not all the others. I’m not sure how much they’ve diverged by now though.

File: src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs (hunk 1 above)
Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs:68]: I've added tests in https://github.com/dfds/capability-service/pull/24/commits/2eb021cf3f323e952618ba40dc9e2f1a96455b85 and https://github.com/dfds/capability-service/pull/24/commits/a8a559a1dbe0667eb51ae80e03240872fd43dab2 for respectively MessagingHandlerNotAvailable and MessagingMessageIncomprehensible

Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs:68]: Regarding porting the implementation, I can't see any reason why not, maybe improve the naming of certain Exceptions.

**Comment by JacobHeidelbachDFDS** [comment]
I believe we need to iterate over all usages of DomainEventRegistry now, and accept this as technical debt that is expensive to touch.

File: src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs (hunk 1 above)
Comment (janwiberg-dfds) [reply src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs:68]: I'm open for the idea but it smells sisyphean to me, copying code to and from the various services. Consider whether time can be better spent on proactive infrastructure work instead maybe?

**Review by janwiberg-dfds — APPROVED** [review]

**Pull request merged**